xarray/core/variable.py

```diff
@@ -1063,20 +1063,30 @@ class Coordinate(Variable):
     """Wrapper around pandas.Index that adds xarray specific functionality.
 
     The most important difference is that Coordinate objects must always have a
-    name, which is the dimension along which they index values.
+    name, which is the dimension along which they index values unless another
+    dimension name is specified.
 
     Coordinates must always be 1-dimensional. In addition to Variable methods
     and properties (attributes, encoding, broadcasting), they support some
     pandas.Index methods directly (e.g., get_indexer), even though pandas does
     not (yet) support duck-typing for indexes.
     """
 
-    def __init__(self, name, data, attrs=None, encoding=None, fastpath=False):
-        super(Coordinate, self).__init__(name, data, attrs, encoding, fastpath)
+    def __init__(self, name, data, attrs=None, encoding=None,
```

Comment:
I would slightly rather change the signature from `name` -> `dims` in the first argument (to match `Variable`), and add `name=None` as an optional parameter at the end (before `fastpath`). This could possibly break use of `Coordinate(name=..., data=...)` but I expect that usage is relatively rare -- this part of the API is usually not touched by users.

Possibly it would be a good idea to rename `Coordinate` -> `IndexVariable`, and change it's intended usage to cover anytime someone wants to represent a `pandas.Index` in xarray. This could be useful for anyone who needs support for custom pandas types like Period or Categorical, even if they won't actually use the array for indexing.